tutor/specs/components/payments/manage.spec.jsx

```diff
@@ -19,8 +20,15 @@ describe('Student Payments Management', () => {
         Router.makePathname.mockImplementation(() => '/foo');
     });
 
-    it('renders and matches snapshot', () => {
-        expect.snapshot(<C noRef><Payments /></C>).toMatchSnapshot();
+    it('renders and matches snapshot', async () => {
+        const component = renderer.create(<C noRef><Payments /></C>)
+
+        // clearing react lifecycle
```

Comment:
@jivey let me know if there is a cleaner way to do this in the test framework that we have set up here

Reply:
Not sure if cleaner, but it looks like we've handled this in other places with a `waitFor` helper that kinda does the same thing:

```
 await waitFor(() => !currentPurchases.api.isPending)
```

Reply:
after the api request completes (which happens in one tick because its faked) you have to wait another tick for the react stuff to update, which is why i have to have two awaits, to clear the promise queue twice. it can actually be made a little better than what i had before by using `setImmediate`. the waitFor helper works, but only because it has a minimum wait of 100ms, which is pretty aggressive, i'm not sure i really want to go around sprinkling that everywhere...